src/main/java/com/synopsys/integration/detect/tool/detector/DetectorRuleFactory.java

```diff
@@ -143,11 +144,17 @@ private DetectorRuleSet createRules(DetectDetectableFactory detectableFactory) {
         ruleSet.addDetector(DetectorType.PACKAGIST, "Composer", ComposerLockDetectable.class, detectableFactory::createComposerDetectable).defaults().build();
 
         DetectorRule<?> pipEnv = ruleSet.addDetector(DetectorType.PIP, "Pip Env", PipenvDetectable.class, detectableFactory::createPipenvDetectable).defaults().build();
+        DetectorRule<?> pipfileLock = ruleSet.addDetector(DetectorType.PIP, "Pipfile Lock", PipfileLockDetectable.class, detectableFactory::createPipfileLockDetectable).defaults()
+            .build();
         DetectorRule<?> pipInspector = ruleSet.addDetector(DetectorType.PIP, "Pip Inspector", PipInspectorDetectable.class, detectableFactory::createPipInspectorDetectable)
             .defaults().build();
         DetectorRule<?> poetry = ruleSet.addDetector(DetectorType.POETRY, "Poetry", PoetryDetectable.class, detectableFactory::createPoetryDetectable).defaults().build();
+
         ruleSet.yield(pipInspector).to(pipEnv);
         ruleSet.yield(poetry).to(pipEnv);
+        ruleSet.yield(pipfileLock).to(poetry);
+        ruleSet.yield(pipfileLock).to(pipEnv);
+        ruleSet.yield(pipfileLock).to(pipInspector);
```

Comment:
Is that true? Do we prefer the graph of pipEnv over the lock file? Does the normal graph have relationships?

Also why is poetry involved? If we have both Poetry and PipEnv files, we probably want both right?

Reply:
Yes PipenvDetectable produces graph with relationships, so I think it's clearly better.

With Poetry, I went back and forth, and could get on board with removing that yield.  I glanced at a few articles touting Poetry as a more accurate manager, and Poetry provides relationships.  But I have to admit I'm not 100% clear on how the two could coexist, so maybe it would be safer to just have results from both.